lib/receiver.js

```diff
@@ -304,7 +366,13 @@ class Receiver extends Writable {
       this._totalPayloadLength += this._payloadLength;
       if (this._totalPayloadLength > this._maxPayload && this._maxPayload > 0) {
         this._loop = false;
-        return error(RangeError, 'Max payload size exceeded', false, 1009);
+        return error(
+          RangeError,
+          'Max payload size exceeded',
+          false,
+          1009,
+          'WS_ERR_INVALID_DATA_PAYLOAD_LENGTH'
```

Comment:
How about `WS_ERR_MESSAGE_TOO_BIG` or something like that? The payload length is not invalid per se.

Reply:
I've renamed the `WS_ERR_INVALID_DATA_PAYLOAD_LENGTH` code to `WS_ERR_UNSUPPORTED_MESSAGE_LENGTH`. That's a bit more consistent with the other codes (`MESSAGE_TOO_BIG` isn't a noun), but it's explicitly a message rather than a single frame, and it's just not supported in this connection rather than completely invalid. Does that work for you?

Reply:
`WS_ERR_UNSUPPORTED_MESSAGE_LENGTH` is perfect to address my previous comment where payload length > 2^53 - 1. However in this case it is not unsupported. The error in this case is that the message length exceeds the maximum allowed message length as defined by the `maxPayload` option.

Reply:
> However in this case it is not unsupported. The error in this case is that the message length exceeds the maximum allowed message length as defined by the maxPayload option.

That's what I mean by 'unsupported'. Messages that go beyond it aren't "invalid" - imo invalid means against the specification, i.e. this will never work on any conformant websocket connection.

With 'unsupported' I'm trying to say that it may be valid, and it might work elsewhere, but it won't work on this connection right now (e.g. due to the configured maxPayload option). That matches standard use elsewhere - e.g. close code 1003 ("unsupported data" - valid data in a format that this server can't process) or HTTP 415 ("unsupported media type" - a valid request with a media type that this server refuses to handle).

I think that fits both this case (unsupported due to ws maxPayload config) and the other case (unsupported due to JS limitations).

I see what you mean about data frame size vs total message size though. How about we split the two cases into:

* `WS_ERR_UNSUPPORTED_MESSAGE_LENGTH` - overall message is too long (> maxPayload)
* `WS_ERR_UNSUPPORTED_DATA_PAYLOAD_LENGTH` - a single data frame is too long (> 2^53-1)

Reply:
> How about we split the two cases into

Works for me and that is what I prefer.

Reply:
Great, done :+1: